xa/java/jms/department-helidon-teq/src/main/java/com/oracle/mtm/sample/Configuration.java

```diff
@@ -0,0 +1,166 @@
+included in all copies or substantial portions of the Software. **
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+package com.oracle.mtm.sample;
+
+import oracle.jakarta.jms.AQjmsFactory;
+import oracle.jakarta.jms.AQjmsSession;
+import oracle.jakarta.jms.AQjmsTextMessage;
+import oracle.jakarta.jms.AQjmsTopicSubscriber;
+import oracle.tmm.common.TrmConfig;
+import oracle.tmm.jta.common.DataSourceInfo;
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+import oracle.ucp.jdbc.PoolXADataSource;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.Initialized;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.jms.JMSException;
+import jakarta.jms.Session;
+import jakarta.jms.Topic;
+import jakarta.jms.TopicConnection;
+import jakarta.jms.TopicSession;
+import jakarta.jms.XATopicConnectionFactory;
+import java.lang.invoke.MethodHandles;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class Configuration {
+
+    private PoolXADataSource xaDataSource;
+    private PoolDataSource dataSource;
+    private Logger logger;
+
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.url")
+    String url;
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.user")
+    String user;
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.password")
+    String password;
+
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.rmid")
+    String rmid;
+
+    private void init(@Observes @Initialized(ApplicationScoped.class) Object event) {
+        initialiseTrmLogger();
+        initialiseDataSource();
+        new Thread(() -> {
+            consumeEvents();
+        }).start();
+
+    }
+
+    private void consumeEvents() {
+        // create a JMS topic connection and session
+        XATopicConnectionFactory tcf = null;
+        try {
+            tcf = AQjmsFactory.getXATopicConnectionFactory(this.getXADatasource());
+            TopicConnection conn = tcf.createXATopicConnection();
+            conn.start();
+            TopicSession session =
+                    (AQjmsSession) conn.createSession(true, Session.AUTO_ACKNOWLEDGE);
+
+            // create a subscriber on the topic
+            Topic topic = ((AQjmsSession) session).getTopic(user, "microtxshqueue2");
```

Comment:
we need to read this config microtxshqueue2 from properties file as we have set the property there

Reply:
Already, exposed it as config element. Now I replaced hardcoded value with config param